frontend/src/metabase/visualizations/visualizations/PivotTable.jsx

```diff
@@ -213,129 +204,131 @@ export default class PivotTable extends Component {
+        width: span * CELL_WIDTH,
+        x: offset * CELL_WIDTH,
+        y: (topHeaderRows - maxDepthBelow - 1) * CELL_HEIGHT,
+      };
+    };
+
     const leftHeaderWidth =
       rowIndexes.length > 0
         ? LEFT_HEADER_LEFT_SPACING + rowIndexes.length * LEFT_HEADER_CELL_WIDTH
         : 0;
 
-    function columnWidth({ index }) {
-      if (topIndex.length === 0) {
-        return CELL_WIDTH;
-      }
-      const indexItem = topIndex[index];
-      return indexItem[indexItem.length - 1].length * CELL_WIDTH;
-    }
-
-    function getSpan(children) {
-      return children.length === 0
-        ? 1
-        : children.reduce((sum, child) => sum + getSpan(child.children), 0);
-    }
-    function rowHeight({ index }) {
-      if (leftIndex.length === 0) {
-        return CELL_HEIGHT;
-      }
-      const span = getSpan(leftIndex[index]);
-      return span * CELL_HEIGHT;
-    }
-
-    // Create three memoized cell renderers
     // These are tied to the `multiLevelPivot` call, so they're awkwardly shoved in render for now
 
-    const topHeaderRenderer = _.memoize(
-      ({ key, style, columnIndex }) => {
-        const rows = topIndex[columnIndex];
-        return (
-          <div key={key} style={style} className="border-bottom border-medium">
-            <div className="flex flex-column px1 full-height justify-end">
-              {rows.map((row, index) => (
-                <Flex style={{ height: CELL_HEIGHT }}>
-                  {row.map(({ value, span }) => (
-                    <div
-                      style={{ width: CELL_WIDTH * span }}
-                      className={cx("flex flex-column justify-center", {
-                        "border-bottom": index < rows.length - 1,
-                      })}
-                    >
-                      <Ellipsified>
-                        {index < topIndexFormatters.length
-                          ? topIndexFormatters[index](value)
-                          : value // Metric names don't have formatters
-                        }
-                      </Ellipsified>
-                    </div>
-                  ))}
-                </Flex>
-              ))}
-            </div>
-          </div>
-        );
-      },
-      ({ columnIndex }) => columnIndex,
-    );
-
-    const leftHeaderRenderer = _.memoize(
-      ({ key, style, index }) => (
-        <div
-          key={key}
-          style={style}
-          className="border-right border-medium bg-light"
-        >
-          {(leftIndex[index] || []).map(item => (
-            <LeftHeaderSection
-              item={item}
-              settings={settings}
-              onUpdateVisualizationSettings={onUpdateVisualizationSettings}
+    const bodyRenderer = ({ key, style, rowIndex, columnIndex }) => (
+      <div key={key} style={style} className="flex">
+        {getRowSection(columnIndex, rowIndex).map(
+          (
+            { value, hasChildren, isSubtotal, isGrandTotal, clicked },
```

Comment:
Is `hasChildren` used anywhere?

Reply:
Ah good catch. `hasChildren` isn't used for body cells.